### PR TITLE
Bug fixes and minor improvements

### PR DESCRIPTION
**Pull request by shreyakhajanchi**
1. Have a save session button on prepare migration page.
2. Table name added in all the schema issues
3. Display all the issues when no table is selected
4. Repharse the default value warning
5. Disabled Add column button till transformations is in place
6. Solved the issue - In datastream details card when test connection button is clicked, there is no indication whether the connection is being tested. It can lead to multiple clicks and in turn multiple test connection calls
7. Marked bulk migration as (Experimental)
8. If migration_shard_id column already exists append some number to it
9. Solved the issue - Drop a column and then when trying to restoring it shows multiple entries with the same name
10. Restoring a dropped column will restore the defaults that were populated initially from the source
11. Fixed the issue - Text form input not working properly for connection profile config
12. Fixed the issue - Download configuration doesn't download on the first click
13. Added an info icon for what will be populated in migration_shard_id column

### CODE REVIEW

File: internal/reports/report_helpers.go [hunk 1]
```diff
@@ -384,7 +384,7 @@ var IssueDB = map[internal.SchemaIssue]struct {
 	severity severity
 	batch    bool // Whether multiple instances of this issue are combined.
 }{
-	internal.DefaultValue:                {Brief: "Some columns have default values which Spanner does not support", severity: warning, batch: true},
+	internal.DefaultValue:                {Brief: "Some columns have default values which HarbourBridge does not migrate. Please add the default constraints manually after the migration is complete", severity: warning, batch: true},
```
Comment (bharadwaj-aditya) [inline internal/reports/report_helpers.go:387]: reduce the severity to note

File: internal/reports/report_helpers.go [hunk 2]
```diff
@@ -411,8 +411,8 @@ var IssueDB = map[internal.SchemaIssue]struct {
 	internal.InterleavedRenameColumn:     {Brief: "Candidate for Interleaved Table", severity: suggestion},
 	internal.InterleavedChangeColumnSize: {Brief: "Candidate for Interleaved Table", severity: suggestion},
 	internal.RowLimitExceeded:            {Brief: "Non key columns exceed the spanner limit of 1600 MB. Please modify the column sizes", severity: errors},
-	internal.ShardIdColumnAdded:          {Brief: "Column migration_shard_id was added because this is a sharded migration and this column cannot be dropped", severity: note},
-	internal.ShardIdColumnPrimaryKey:     {Brief: "Column migration_shard_id is not a part of primary key. You may go to the Primary Key tab and add this column as a part of Primary Key", severity: suggestion},
+	internal.ShardIdColumnAdded:          {Brief: "column was added because this is a sharded migration and this column cannot be dropped", severity: note},
```
Comment (bharadwaj-aditya) [inline internal/reports/report_helpers.go:414]: How will the customer know which column was added ?

File: ui/src/app/components/object-detail/object-detail.component.ts [hunk 3]
```diff
@@ -260,24 +262,24 @@ export class ObjectDetailComponent implements OnInit {
...
+              srcOrder: new FormControl(col.srcOrder),
+              srcColName: new FormControl(col.srcColName),
+              srcDataType: new FormControl(col.srcDataType),
+              srcIsPk: new FormControl(col.srcIsPk),
+              srcIsNotNull: new FormControl(col.srcIsNotNull),
+              srcColMaxLength: new FormControl(col.srcColMaxLength),
+              spOrder: new FormControl(col.srcOrder),
+              spColName: new FormControl(col.srcColName),
+              spDataType: new FormControl(
+                this.defaultTypeMap[col.srcDataType].Name
```
Comment (bharadwaj-aditya) [inline ui/src/app/components/object-detail/object-detail.component.ts:276]: Where is this populated ?

File: ui/src/app/components/prepare-migration/prepare-migration.component.ts [hunk 4]
```diff
@@ -844,5 +852,22 @@ export class PrepareMigrationComponent implements OnInit {
...
+    this.sidenav.openSidenav()
+    this.sidenav.setSidenavComponent('saveSession')
+    this.sidenav.setSidenavDatabaseName(this.conv.DatabaseName)
+  }
+  downloadSession() {
+    var a = document.createElement('a')
+    // JS automatically converts the input (64bit INT) to '9223372036854776000' during conversion as this is the max value in JS.
+    // However the max value received from server is '9223372036854775807'
+    // Therefore an explicit replacement is necessary in the JSON content in the file.
+    let resJson = JSON.stringify(this.conv).replace(/9223372036854776000/g, '9223372036854775807')
```
Comment (bharadwaj-aditya) [inline ui/src/app/components/prepare-migration/prepare-migration.component.ts:866]: We should move this to a common util method somewhere

File: ui/src/app/components/summary/summary.component.html [hunk 5]
```diff
@@ -118,29 +118,8 @@
         </svg>
       </div>
       <div class="no-issue-message">
-        Woohoo! No issues or suggestions<br />found for {{ currentObject.name }}.
+        Woohoo! No issues or suggestions<br />found.
       </div>
     </div>
   </div>
-  <div class="no-object-container" *ngIf="!currentObject">
```
Comment (bharadwaj-aditya) [inline ui/src/app/components/summary/summary.component.html:125]: Why is this removed ? Will this never be triggered ?

**Review by bharadwaj-aditya — CHANGES_REQUESTED** [review]

File: webv2/config.json [hunk 6]
```diff
@@ -1,4 +1,4 @@
 {
- "GCPProjectID": "",
- "SpannerInstanceID": ""
+ "GCPProjectID": "span-cloud-testing",
```
[the remark this answers is not in the document]
Comment (shreyakhajanchi) [reply webv2/config.json:2]: yes, it was a mistake will remove this file changes

File: ui/src/app/components/summary/summary.component.html (hunk 5 above)
Comment (shreyakhajanchi) [reply ui/src/app/components/summary/summary.component.html:125]: this was been shown when no table was selected, after this code change if no table is selected then will show all the issues

File: internal/reports/report_helpers.go (hunk 2 above)
Comment (shreyakhajanchi) [reply internal/reports/report_helpers.go:414]: We append specific information like table name, column name to these briefs before printing out

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM
